core/src/block.rs

```diff
@@ -918,33 +918,30 @@ impl VersionedCandidateCommittedBlock {
         &self.as_v1().header
     }
 
-    /// Return the signatures (as `payload`) that are verified with the `hash` of this block.
-    #[inline]
-    pub fn verified_signatures(&self) -> impl Iterator<Item = &SignatureOf<Self>> {
-        self.as_v1()
-            .verified_signatures()
-            .map(SignatureOf::transmute_ref)
-    }
-
     /// Revalidate transaction hashes, verify signatures and produce [`VersionedCommittedBlock`]
     ///
     /// # Errors
     /// - If transaction hashes don't match the hashes stored in the block header
     /// - If signatures verification fails
-    pub fn revalidate(self, topology: &Topology) -> Result<VersionedCommittedBlock, eyre::Report> {
+    pub fn revalidate(
+        self,
+        topology: &Topology,
+    ) -> Result<VersionedCommittedBlock, (Self, eyre::Report)> {
         self.into_v1()
             .revalidate(topology)
             .map(VersionedCommittedBlock::from)
+            .map_err(|err| (err.0.into(), err.1))
     }
 
     /// Revalidate transaction hashes and produce [`VersionedCommittedBlock`]
     ///
     /// # Errors
     /// - If transaction hashes don't match the hashes stored in the block header
-    pub fn revalidate_hashes(self) -> Result<VersionedCommittedBlock, eyre::Report> {
+    pub fn revalidate_hashes(self) -> Result<VersionedCommittedBlock, (Self, eyre::Report)> {
         self.into_v1()
             .revalidate_hashes()
             .map(VersionedCommittedBlock::from)
+            .map_err(|err| (err.0.into(), err.1))
```

Comment:
Maybe we need to provide this information some other way. A new type perhaps? Tuple struct?

Reply:
I'd rather not